index.bs

```diff
@@ -3830,6 +3859,802 @@ opened</dfn> steps given |window|, |type| and |message|.
+
+<div algorithm>
+To <dfn>get the response content info</dfn> given |response|.
+
+1. Return a new map matching the <code>network.ResponseContent</code>
+   production, with the <code>size</code> field set to |response|'s [=response body
+   info=]'s [=decoded size=]
+
+</div>
+
+#### The network.ResponseData type #### {#type-network-ResponseData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.ResponseData = {
+    url: text,
+    protocol: text,
+    status: js-uint,
+    statusText: text,
+    fromCache: bool,
+    headers: [*network.Header],
+    mimeType: text,
+    bytesReceived: js-uint,
+    headersSize: js-uint / null,
+    bodySize: js-uint / null,
+    content: network.ResponseContent
+};
+</pre>
+
+The <code>network.ResponseData</code> type represents the response to a network
+request.
+
+<div algorithm>
+
+To <dfn>get the protocol</dfn> given |response|:
+
+1. Let |protocol| be the empty string.
+
+1. If |response|'s [=final connection timing info=] is not null, set |protocol|
+   to |response|'s [=final connection timing info=]'s [=ALPN negotiated
+   protocol=].
+
+1. If |protocol| is the empty string, or is equal to "<code>unknown</code>":
+
+  1. Set |protocol| to |response|'s [=response/url=]'s [=url/scheme=]
+
+  1. If |protocol| is equal to either "<code>http</code>" or
+     "<code>https</code>" and |response| has an associated HTTP Response.
+
+     Note: [[FETCH]] isn't clear about the relation between a HTTP network response and
+     a response object.
+
+     <!-- It would be better to move this into Fetch itself. -->
+
+    1. Let |http version| be the HTTP Response's Status line's HTTP-version [[HTTP11]].
+
+    1. If |http version| starts with "<code>HTTP/</code>":
+
+      1. Let |version| be the [=code unit substring=] of |http version| from 5
+         to |http version|'s [=string/length=].
+
+      1. If |version| is "<code>0.9</code>", set |protocol| to
+         "<code>http/0.9</code>", otherwise if |version| is "<code>1.0</code>",
+         set |protocol| to "<code>http/1.0</code>", otherwise if |version| is
+         "<code>1.1</code>", set |protocol| to "<code>http/1.1</code>".
+
+1. Return |protocol|.
+
+</div>
+
+<div algorithm>
+To <dfn>get the response data</dfn> given |response|:
+
+1. Let |url| be the result of running the [=URL serializer=] with |response|'s
+   [=URL=].
+
+1. Set |protocol| to [=get the protocol=] given |response|.
+
+1. Let |status| be |response|'s [=response/status=].
+
+1. Let |status text| be |response|'s [=status message=].
+
+1. If |response|'s [=cache state=] is "<code>local</code>", let |from cache| be
+   true, otherwise let it be false.
+
+1. Let |headers| be an empty list.
+
+1. Let |mime type| be the [=computed mime type=] for |response|.
+
+   Note: this is whatever MIME type the browser is actually using, even if it
+   isn't following the exact algorithm in the [[MIMESNIFF]] specification.
```

Comment:
Cf discussion at https://phabricator.services.mozilla.com/D166373#inline-918466 about including or not the (optional) charset in the `mime type`. Following the `computed mime type` algorithm, the charset should in theory be part of the `mime type` but since CDP does not include it, we might do the same.

I guess what we want here is the https://mimesniff.spec.whatwg.org/#mime-type-essence ?